src/js/controller/controller.js

```diff
@@ -977,10 +977,18 @@ Object.assign(Controller.prototype, {
             _programController.controls = mode;
         };
 
+        this.addCues = function (cues) {
+            const existingCues = this.getCues();
+            const newCues = existingCues ? existingCues.concat(cues) : cues;
```

Comment:
1. Let's default the cues in model to an empty array so that we don't need this ternary operator.
2. In `setCues` let's validate the input - it must be an array, otherwise we should not update the model.